src/main/java/gregtech/integration/jei/GTJeiPlugin.java

```diff
@@ -127,37 +132,43 @@ public void register(IModRegistry registry) {
-        
+
+        registry.addIngredientInfo(Objects.requireNonNull(Materials.Air.getFluid(1000)), VanillaTypes.FLUID, I18n.format("gregtech.machine.air_collector.jei_description"));
+
         String primitiveBlastId = GTValues.MODID + ":" + "primitive_blast_furnace";
         registry.addRecipes(RecipeMaps.PRIMITIVE_BLAST_FURNACE_RECIPES.stream()
             .map(PrimitiveBlastRecipeWrapper::new)
             .collect(Collectors.toList()), primitiveBlastId);
         registry.addRecipeCatalyst(MetaTileEntities.PRIMITIVE_BLAST_FURNACE.getStackForm(), primitiveBlastId);
-        
+
         String cokeOvenId = GTValues.MODID + ":" + "coke_oven";
         registry.addRecipes(RecipeMaps.COKE_OVEN_RECIPES.stream()
             .map(CokeOvenRecipeWrapper::new)
             .collect(Collectors.toList()), cokeOvenId);
         registry.addRecipeCatalyst(MetaTileEntities.COKE_OVEN.getStackForm(), cokeOvenId);
-        
-        List<OreByProduct> oreByproductList = new CopyOnWriteArrayList<OreByProduct>();
+
+        List<OreByProduct> oreByproductList = new CopyOnWriteArrayList<>();
         for (Material material : Material.MATERIAL_REGISTRY) {
             if (material instanceof DustMaterial && OreDictUnifier.get(OrePrefix.ore, material) != ItemStack.EMPTY) {
-            	 oreByproductList.add(new OreByProduct((DustMaterial)material));
+                oreByproductList.add(new OreByProduct((DustMaterial) material));
             }
         }
         String oreByProductId = GTValues.MODID + ":" + "ore_by_product";
         registry.addRecipes(oreByproductList, oreByProductId);
-        for(MetaTileEntity machine : MetaTileEntities.MACERATOR)
-        	registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);        
-        for(MetaTileEntity machine : MetaTileEntities.ORE_WASHER)
-        	registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);        
-        for(MetaTileEntity machine : MetaTileEntities.CENTRIFUGE)
-        	registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);        
-        for(MetaTileEntity machine : MetaTileEntities.THERMAL_CENTRIFUGE)
-        	registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);
+        for (MetaTileEntity machine : MetaTileEntities.MACERATOR)
+            registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);
+        for (MetaTileEntity machine : MetaTileEntities.ORE_WASHER)
+            registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);
+        for (MetaTileEntity machine : MetaTileEntities.CENTRIFUGE)
+            registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);
+        for (MetaTileEntity machine : MetaTileEntities.THERMAL_CENTRIFUGE)
+            registry.addRecipeCatalyst(machine.getStackForm(), oreByProductId);
+
+        registry.addRecipes(IntCircuitRecipeWrapper.create(), IntCircuitCategory.UID);
```

Comment:
Is it possible to move it up so it will get registered before other recipes? As it would be better to get postilion up front in JEI.

Reply:
The last commit ought to have done it.